project_restorer/src/main/scala/PlutoCoreMessageProcessor.scala

```diff
@@ -36,28 +52,33 @@ class PlutoCoreMessageProcessor(mxsConfig:MatrixStoreConfig)(implicit mat:Materi
       .run()
   }
 
-
-  def searchAssociatedMedia(projectId: Int, vault: Vault)  = {
-   for {
-     result <- filesByProject(vault, projectId.toString)
-   } yield result
+  def getNearLineResults(projectId: Int) = {
+    matrixStoreBuilder.withVaultFuture(mxsConfig.nearlineVaultId) { vault =>
+      searchAssociatedNearlineMedia(projectId, vault).map(Right.apply)
+    }
   }
 
+  def getOnlineResults(projectId: Int) = {
+    searchAssociatedOnlineMedia(projectId, vidispineCommunicator).map(Right.apply)
+  }
 
-  def handleStatusMessage(updateMessage: ProjectUpdateMessage, routingKey: String, framework: MessageProcessingFramework):Future[Either[String, Json]] = {
-       matrixStoreBuilder.withVaultFuture(mxsConfig.nearlineVaultId) {vault =>
-        searchAssociatedMedia(updateMessage.id, vault).map(results=> {
-
-          if(results.length < 10000 ){
-            framework.bulkSendMessages(routingKey, results)
-            val msg = RestorerSummaryMessage(updateMessage.id, ZonedDateTime.now(), updateMessage.status, results.length, 0)
-            Right(msg.asJson)
-          }
-          else {
-            throw new RuntimeException("Too many files attached to project")
+  def handleStatusMessage(updateMessage: ProjectUpdateMessage, routingKey: String, framework: MessageProcessingFramework): Future[Either[String, MessageProcessorReturnValue]] = {
+    Future.sequence(Seq(getNearLineResults(updateMessage.id), getOnlineResults(updateMessage.id))).map(allResults => {
+      (allResults.head, allResults(1)) match {
+        case (Right(nearlineResults), Right(onlineResults)) =>
+          if (nearlineResults.length < 10000 && onlineResults.length < 10000) {
+            framework.bulkSendMessages(routingKey, nearlineResults)
+            framework.bulkSendMessages(routingKey, onlineResults)
+            val msg = RestorerSummaryMessage(updateMessage.id, ZonedDateTime.now(), updateMessage.status, numberOfAssociatedFilesNearline = nearlineResults.length, numberOfAssociatedFilesOnline = onlineResults.length)
+            Right(MessageProcessorConverters.contentToMPRV(msg.asJson))
+          } else {
+            throw new RuntimeException(s"Too many files attached to project ${updateMessage.id}, nearlineResults = ${nearlineResults.length}, onlineResults = ${onlineResults.length}")
           }
-        })
+        case (Left(nearlineErr), _)=>
+          logger.error("Could not connect to Matrix store: $onlineErr")
+          Left(nearlineErr)
       }
+    })
```

Comment:
The compiler may complain that the `match` will fail if the getOnlineResults returns a `Left`. Now, we know of course that it won't; but it might be worth just putting a simple `Left` clause in here for that (a) so we shut the compiler up and (b) in case anyone in the future does return a Left, assuming (reasonably) that it should work as it would actually crash the processor and dump the message to DLQ.